plugins/woocommerce/tests/legacy/unit-tests/order/class-wc-tests-crud-orders.php

```diff
@@ -611,6 +611,190 @@ public function test_calculate_shipping() {
+		$zone_1->set_zone_order( 1 );
+		$zone_1->add_location( 'US:NY', 'state' );
+		$instance_id_1 = $zone_1->add_shipping_method( 'flat_rate' );
+		$zone_1->save();
+
+		$zone_2 = new WC_Shipping_Zone();
+		$zone_2->set_zone_name( 'Florida' );
+		$zone_2->set_zone_order( 2 );
+		$zone_2->add_location( 'US:FL', 'state' );
+		$instance_id_2 = $zone_2->add_shipping_method( 'flat_rate' );
+		$zone_2->save();
+
+		$zone_3 = new WC_Shipping_Zone();
+		$zone_3->set_zone_name( 'California and Illinois' );
+		$zone_3->set_zone_order( 3 );
+		$zone_3->add_location( 'US:CA', 'state' );
+		$zone_3->add_location( 'US:IL', 'state' );
+		$instance_id_3 = $zone_3->add_shipping_method( 'flat_rate' );
+		$instance_id_4 = $zone_3->add_shipping_method( 'flat_rate' );
+		$zone_3->save();
+
+		$map = array(
+			$shipping_class_1['term_id'] => $instance_id_1, // NY
+			$shipping_class_2['term_id'] => $instance_id_2, // FL
+			$shipping_class_3['term_id'] => $instance_id_3, // CA
+			$shipping_class_4['term_id'] => $instance_id_4, // IL
+		);
+
+		// 2 every flat rate add no cost by default and configure the cost for only one of the shipping classes. Different to each flat rate.
+		foreach ( $map as $shipping_class_id => $instance_id ) {
+			$data = array(
+				'woocommerce_flat_rate_title'                                      => 'Flat rate',
+				'woocommerce_flat_rate_tax_status'                                 => 'taxable',
+				'woocommerce_flat_rate_cost'                                       => '',
+				'woocommerce_flat_rate_class_cost_' . $shipping_class_1['term_id'] => '',
+				'woocommerce_flat_rate_class_cost_' . $shipping_class_2['term_id'] => '',
+				'woocommerce_flat_rate_class_cost_' . $shipping_class_3['term_id'] => '',
+				'woocommerce_flat_rate_class_cost_' . $shipping_class_4['term_id'] => '',
+				'woocommerce_flat_rate_type'                                       => 'class',
+				'instance_id'                                                      => $instance_id,
+			);
+
+			$data['woocommerce_flat_rate_class_cost_' . $shipping_class_id ] = $shipping_class_id * 10;
+
+			$shipping_method = WC_Shipping_Zones::get_shipping_method( $instance_id );
+			$shipping_method->set_post_data( wp_unslash( $data ) ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			$_REQUEST['instance_id'] = $instance_id;
+			$shipping_method->process_admin_options();
+		}
+
+		WC_Cache_Helper::get_transient_version( 'shipping', true );
+		// Set up has finished. Now we need to imitate the different scenarios.
+
+		// Set customer location to US. NY by default.
+		WC_Helper_Shipping::force_customer_us_address();
+
+		WC()->cart->empty_cart();
+
+		WC()->cart->add_to_cart( $product_1->get_id(), 1 );
+
+		WC()->cart->calculate_shipping();
+
+		// Customers address in in NY and product_1 is shippable to NY.
+		$this->assertEquals( (string) ( 10 * $shipping_class_1['term_id'] ), WC()->cart->get_shipping_total() );
```

Comment:
(3/3) Finally, and building on the earlier 2 comments in the series, this lets us avoid the arithmetic at this point:

```php
$this->assertEquals( $method_1_cost, WC()->cart->get_shipping_total() );
```

Additionally and optionally, for this and other assertions we could move the comment into the description string:

```php
$this->assertEquals( 
	$method_1_cost, 
	WC()->cart->get_shipping_total(),
	'Customers address in in NY and product_1 is shippable to NY.'
);